.github/workflows/_publish-docs-on-release.yml

```diff
@@ -35,7 +35,7 @@ jobs:
       - name: Check out ${{ inputs.project }}
         uses: actions/checkout@v4
         with:
-          fetch-depth: 0  # Fetches branches and tags
+          ref: main  # Fetch the main branch with CHANGELOG.rst modified during full release
```

Comment:
Problem 2. 

I found the source of the problem why changelog didn't show the latest version upon release. 

When push a tag (commit) to `upstream`, the tag does not contain updated CHANGELOG.rst. The CHANGELOG.rst is modified during the CI which has an additional commit in `main`.

Hence, when we build a doc, we should checkout `main` instead of the `tag`!